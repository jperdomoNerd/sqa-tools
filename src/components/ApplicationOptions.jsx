import React from 'react';

const ApplicationOptions = ({ application }) => {
  if (application === 'POS Legacy') {
    return (
      <React.Fragment>
        <option value="sale">sale</option>
        <option value="achcredit">achcredit</option>
        <option value="achdebit">achdebit</option>
        <option value="activate">activate</option>
        <option value="auth">auth</option>
        <option value="createtoken">createtoken</option>
        <option value="credit">credit</option>
        <option value="ebppinvoice">ebppinvoice</option>
        <option value="force">force</option>
        <option value="guarantee">guarantee</option>
        <option value="inquire">inquire</option>
        <option value="moto">moto</option>
        <option value="reauth">reauth</option>
        <option value="recurringauth">recurringauth</option>
        <option value="recurringcredit">recurringcredit</option>
        <option value="recurringsale">recurringsale</option>
        <option value="reload">reload</option>
        <option value="repeatauth">repeatauth</option>
        <option value="repeatsale">repeatsale</option>
        <option value="return">return</option>
        <option value="reversal">reversal</option>
        <option value="salemoto">salemoto</option>
        <option value="signature">signature</option>
        <option value="tokencheck">tokencheck</option>
        <option value="usechecktoken">usechecktoken</option>
        <option value="voiceauth">voiceauth</option>
        <option value="void">void</option>
      </React.Fragment>
    );
  } if (application === 'Cashiering Legacy') {
    return (
      <React.Fragment>
        <option value="sale">sale</option>
        <option value="auth">auth</option>
        <option value="cashoperations">cashoperations</option>
        <option value="credit">credit</option>
        <option value="depositslip">depositslip</option>
        <option value="return">return</option>
      </React.Fragment>
    );
  } if (application === 'POS Reports Legacy') {
    return (
      <React.Fragment>
        <option value="reprint">reprint</option>
        <option value="batchreport">batchreport</option>
        <option value="cashsummary">cashsummary</option>
        <option value="checksummary">checksummary</option>
        <option value="ebppinvoice">ebppinvoice</option>
        <option value="transactions">transactions</option>
        <option value="reports">reports</option>
      </React.Fragment>
    ); 
  } if (application === 'Bridge API') {
    return (
      <React.Fragment>
        <option value="GetGiftCardSwipe">GetGiftCardSwipe</option>
        <option value="Disclaimer">Disclaimer</option>
        <option value="GetSignature">GetSignature</option>
      </React.Fragment>
    );
  } if (application === 'Webpay') {
    return (
      <React.Fragment>
        <option value="sale">sale</option>
        <option value="sale19">sale19</option>
        <option value="auth">auth</option>
        <option value="auth19">auth19</option>
        <option value="createtoken">createtoken</option>
        <option value="createtoken19">createtoken19</option>
      </React.Fragment>
    );
  }
  return (<React.Fragment></React.Fragment>);
}

export default ApplicationOptions;
