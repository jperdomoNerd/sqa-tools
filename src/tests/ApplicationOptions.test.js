import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ApplicationOptions from '../components/ApplicationOptions';

describe('ApplicationsOptions Component', () => {
  it('<ApplicationOptions /> renders correctly', () => {
    const tree = renderer.create(<ApplicationOptions application='Webpay' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Only show three options', () => {
    render(<ApplicationOptions application='Bridge API' />);
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(3);
  });
  test('Show 27 options', () => {
    render(<ApplicationOptions application='POS Legacy' />);
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(27);
  });
  test('Show 27 options', () => {
    render(<ApplicationOptions application='Webpay' />);
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(6);
  });
  test('Show 27 options', () => {
    render(<ApplicationOptions application='Webpay' />);
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(6);
  });
});
