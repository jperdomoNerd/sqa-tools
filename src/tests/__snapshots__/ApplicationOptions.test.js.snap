// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationsOptions Component <ApplicationOptions /> renders correctly 1`] = `
Array [
  <option
    value="sale"
  >
    sale
  </option>,
  <option
    value="sale19"
  >
    sale19
  </option>,
  <option
    value="auth"
  >
    auth
  </option>,
  <option
    value="auth19"
  >
    auth19
  </option>,
  <option
    value="createtoken"
  >
    createtoken
  </option>,
  <option
    value="createtoken19"
  >
    createtoken19
  </option>,
]
`;
